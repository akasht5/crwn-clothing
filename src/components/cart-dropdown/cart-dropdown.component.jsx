import React from 'react'

import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import { selectCartItems } from '../../redux/cart/cart.selectors'
import { createStructuredSelector } from 'reselect'
import { toggleCartHidden } from '../../redux/cart/cart.actions'
import CartItem from '../cart-item/cart-item.component'
import { CartDropdownContainer,CartItemsContainer,CheckoutButton,EmptyMessageContainer } from './cart-dropdown.styles'
 
const CartDropdown = ({ cartItems,history,dispatch }) => {
    return (
        <CartDropdownContainer> 
            <CartItemsContainer>
                {
                    cartItems.length ?
                    (
                        cartItems.map(cartItem => (
                            <CartItem key={cartItem.id} cartItem={cartItem} />
                        ))
                    ) : (
                        <EmptyMessageContainer>Your Cart is empty !</EmptyMessageContainer>
                    )
                }
            </CartItemsContainer>
            <CheckoutButton onClick={ () => {
                history.push('/checkout');
                dispatch(toggleCartHidden());
            } 
            }>Checkout</CheckoutButton>
        </CartDropdownContainer>
    )
}

const mapStateToProps = createStructuredSelector({
    cartItems : selectCartItems
});

export default withRouter(connect(mapStateToProps)(CartDropdown))
