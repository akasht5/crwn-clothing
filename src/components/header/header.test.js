import React from 'react'
import { shallow, mount } from 'enzyme'
import Header from './header.component'
import { CartDropdown } from '../cart-dropdown/cart-dropdown.component'
import { store } from '../../redux/store'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

describe('<Header /> component',() => {
    let wrapper;

    beforeEach(() => {
       const mockProps = {
           hidden: false
       }

       wrapper = mount(
        <BrowserRouter>
            <Provider store={store}>
            <Header {...mockProps} />
        </Provider>
        </BrowserRouter>
       ); 
    })

    it('Snapshot testing header',() => {
        expect(wrapper).toMatchSnapshot();
    })

    it('Checks for cartDropdown based on hidden state',() => {
        const mockProps = {
            hidden : false
        }
        wrapper = shallow(
            <BrowserRouter>
                <Provider store={store}>
                    <Header {...mockProps} />
            </Provider>
            </BrowserRouter>
           ); 

        const container = wrapper.find(`[data-test='cartdropdown']`);
        console.log(container);
        expect(container.length).toBe(1);
;    })

});