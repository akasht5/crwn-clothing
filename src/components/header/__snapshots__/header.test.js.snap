// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component Snapshot testing header 1`] = `ReactWrapper {}`;
